introspection-engine/connectors/sql-introspection-connector/src/defaults.rs

```diff
@@ -113,3 +119,42 @@ fn set_default(mut default: dml::DefaultValue, column: sql::ColumnWalker<'_>) ->
 fn is_sequence(column: sql::ColumnWalker<'_>) -> bool {
     column.is_single_primary_key() && matches!(&column.default(), Some(d) if d.is_sequence())
 }
+
+fn maybe_prisma1_default(column: sql::ColumnWalker<'_>, ctx: &mut Context<'_>) -> Option<dml::DefaultValue> {
+    let model_and_field = || crate::warnings::ModelAndField {
+        model: ctx.table_prisma_name(column.table().id).prisma_name().into_owned(),
+        field: ctx.column_prisma_name(column.id).prisma_name().into_owned(),
+    };
+
+    if ctx.sql_family().is_postgres() {
+        let native_type: &PostgresType = column.column_type().native_type.as_ref()?.downcast_ref();
+
+        if native_type == &PostgresType::VarChar(Some(25)) {
+            ctx.prisma_1_cuid_defaults.push(model_and_field());
+            return Some(default_cuid());
+        } else if native_type == &PostgresType::VarChar(Some(36)) {
+            ctx.prisma_1_uuid_defaults.push(model_and_field());
+            return Some(default_uuid());
+        }
+    } else if ctx.sql_family().is_mysql() {
+        let native_type: &MySqlType = column.column_type().native_type.as_ref()?.downcast_ref();
+
+        if native_type == &MySqlType::Char(25) {
+            ctx.prisma_1_cuid_defaults.push(model_and_field());
+            return Some(default_cuid());
+        } else if native_type == &MySqlType::Char(36) {
+            ctx.prisma_1_uuid_defaults.push(model_and_field());
+            return Some(default_uuid());
+        }
+    };
```

Comment:
This is more or less just moved. I'd replace that with match expressions if there was more time.